modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py

```diff
@@ -76,30 +76,50 @@ class IgniteSpec(metaclass=ABCMeta):
     This class is a basic Spec
     """
 
-    def __init__(self, service, jvm_opts, full_jvm_opts):
+    def __init__(self, service, jvm_opts=None, full_jvm_opts=None):
+        """
+        Only one of the jvm_opts or full_jvm_opts may be passed.
+
+        :param service: Service
+        :param jvm_opts: If passed will be added or overwrite the default ones.
+        :param full_jvm_opts: If passed will be used 'as is'. None of the default ones will be applied.
+        """
+
+        assert (jvm_opts and not full_jvm_opts) or \
+               (not jvm_opts and full_jvm_opts) or \
+               (not jvm_opts and not full_jvm_opts)
+
         self.service = service
+        self.external_jvm_opts = jvm_opts
+        self.external_full_jvm_opts = full_jvm_opts
 
-        if full_jvm_opts:
-            self.jvm_opts = full_jvm_opts
+    def get_default_jvm_opts(self):
```

Comment:
May be this method should be underscored (hidden)?

Reply:
Yes, will change